configEcho.js

```diff
@@ -0,0 +1,114 @@
+const config = require( './config.js' );
+
+const BASE_URL = process.env.MW_SERVER;
+
+/**
+ * Be careful when expanding the list of tests here.
+ * Each scenario represents 8 scenarios:
+ * - Vector skin (mobile + tablet + desktop + desktop wide)
+ * - Vector 2022 skin (mobile + tablet + desktop + desktop wide)
+ * - Minerva skin (mobile + tablet + desktop + desktop wide)
+ */
+const baseScenarios = [
+	{
+		label: 'Echo smoke test with 0 notifications',
+		path: '/wiki/Test',
+		hashtags: []
+	},
+	{
+		label: 'Echo smoke test with 1 notifications',
+		path: '/wiki/Test',
+		hashtags: [ '#echo-1' ]
+	},
+
+	{
+		label: 'Echo smoke test with 1 seen notifications',
+		path: '/wiki/Test',
+		hashtags: [ '#echo-1-seen' ]
+	},
+	{
+		label: 'Echo smoke test with 99+ notifications',
+		path: '/wiki/Test',
+		hashtags: [ '#echo-100' ]
+	},
+	// Important: Don't use the #echo-drawer tag on echo-100 or echo-1,
+	// doing so will trigger "seen" status and change the color of the notification
+	// icon from red to gray.
+	{
+		label: 'Echo smoke test with 0 notifications',
+		path: '/wiki/Test',
+		hashtags: [ '#echo-drawer' ]
+	}
+];
+
+/**
+ * Creates a list of tests with the given query string and hashtags.
+ *
+ * @param {Array} hashtags additional hashtags
+ * @param {string} queryString to add to URL.
+ * @param {Object} additionalConfig
+ * @return {Array} of scenarios.
+ */
+const makeScenarios = ( hashtags, queryString, additionalConfig = {} ) => {
+	return ( test ) => {
+		const hashtagString = hashtags.concat( [ '#logged-in', '#echo' ] ).concat(
+			test.hashtags
+		).join( ' ' );
+		return Object.assign( {
+			selectors: [ 'viewport' ]
```

Comment:
I'm seeing the following with the `viewport` selector`:
![before](https://user-images.githubusercontent.com/3496279/171449554-13204f52-bb78-4cb6-b91e-d20f6523965d.png)

Changing the selector to `.client-js` (on the html element) leads to:

![MediaWiki_Echo_smoke_test_with_0_notifications_vector-2022_logged-in_echo_echo-drawer_0_client-js_1_tablet](https://user-images.githubusercontent.com/3496279/171450173-6f57c3ea-978a-471d-8b6d-82fa010f6cf5.png)



I think what is happening is a mix of puppeteer's `captureBeyondScreenshot` option being `true` [1] (the default in backstop and puppeteer) which leads to a resize of the window before the screenshot and echo/ooui not handling the resize correctly.


[1] https://pptr.dev/#?product=Puppeteer&version=v14.1.2&show=api-pagescreenshotoptions

Reply:
Fixed in d94d0eabab34969